### ISSUE TITLE
Add constraints on which slots should be documented with which mapping justification

### ISSUE THREAD

**Issue by matentzn** [mapping-commons/sssom#157]
Right now, the core SSSOM model is

subject, predicate, object, mapping justification.

In our flat model, we have not constrained how specific elements, like `subject_match_field` should appear in conjunction with which `mapping justification`. 

<details><summary>Here is a dump that we should probably constrain in one form or another:</summary>
mapping activity:
  abstract: true
  description: A mapping activity is an activity that confirms or provides evidence towards the truthfullness of a mapping.
  slots:
    - confidence
    - author
    - creator
    - mapping_date
    - see_also
    - other
    - comment
derivation:
  is_a: mapping activity
  description: A derivation is a mapping activity that derives a mapping from another mapping source, such as a mapping table or ontology. This process often involves "guessing" the intended mapping predicate. It is recommended that this is well documented.
  slots:
    - mapping_provider
automated match:
  abstract: true
  is_a: match
  description: An automated match is a match which is performed by a software agent, such as an Ontolgy Mapping Tool.
  slots:
    - mapping_tool
    - mapping_tool_version
lexical match:
  is_a: automated match
  description: Lexical match
  slots:
    - subject_match_field
    - object_match_field
    - match_string
    - subject_preprocessing
    - object_preprocessing
cross-walk:
  is_a: automated match
  description: A cross-walk is a match type based on one or more mappings
    - mapping_ids
human curated mapping:
  is_a: mapping activity
  description: Match based on human expert opinion
  slots:
    - curation rule
semantic similarity match:
  slots:
    - semantic_similarity_score
    - semantic_similarity_measure
</details>

**Comment by jonquet** [comment]
Is this related to #146 ?

**Comment by matentzn** [comment]
Yes, seems to be the same. Should merge the two issues in the next round. Thanks for spotting!
